src/files/rename-prompt/RenamePrompt.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import PencilIcon from '../../icons/StrokePencil'
+import CancelIcon from '../../icons/GlyphSmallCancel'
+import Button from '../../components/button/Button'
+
+class RenamePrompt extends React.Component {
+  static propTypes = {
+    cancel: PropTypes.func.isRequired,
+    action: PropTypes.func.isRequired,
+    filename: PropTypes.string.isRequired,
+    folder: PropTypes.bool
+  }
+
+  static defaultProps = {
+    className: '',
+    folder: false
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = { filename: props.filename }
+
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+    this.handleKeyPress = this.handleKeyPress.bind(this)
+  }
+
+  handleChange (event) {
+    this.setState({ filename: event.target.value })
+  }
+
+  handleSubmit (event) {
+    this.props.action(this.state.filename)
+    event.preventDefault()
+  }
+
+  handleKeyPress = (event) => {
+    if (event.key === 'Enter') {
+      this.handleSubmit(event)
+    }
+  }
+
+  handleCancel = (event) => {
+    this.props.cancel()
+    event.preventDefault()
+  }
+
+  render () {
+    let {cancel, className, action, filename, folder, ...props} = this.props
+    className = `${className} w-80 shadow-4 sans-serif relative`
+
+    return (
+      <form className={className} style={{maxWidth: '30em'}} {...props}>
```

Comment:
if you move the onSubmit handler to the form we should get "enter to submit" behaviour from the browser for free!